### PR TITLE
added snowflake queries

### PR DESCRIPTION
**Pull request by robertainslie**
Added Snowflake Queries - merging into the programmable automation branch

### CODE REVIEW

File: hubspot-snowflake-aws-ml-insights/snowflake-queries/snowflake-to-s3-hs-contacts.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        TO_VARCHAR(TO_NUMBER(c.PROPERTY_AGE)) AS AGE,
+        c.PROPERTY_GENDER AS GENDER,
+        c.PROPERTY_FIRSTNAME AS FIRSTNAME,
+        c.PROPERTY_LASTNAME AS LASTNAME,
+        c.PROPERTY_EMAIL AS EMAIL
+    FROM
+    HUB_23826378_DB.V2_DAILY.OBJECTS_CONTACTS c
+    WHERE LENGTH(c.PROPERTY_EXTERNAL_ID) > 0
+  )
+  CREDENTIALS = (AWS_KEY_ID=process.env.AWS_KEY_ID AWS_SECRET_KEY=process.env.AWS_SECRET_KEY)
```
Comment (steven-miller) [inline hubspot-snowflake-aws-ml-insights/snowflake-queries/snowflake-to-s3-hs-contacts.sql:24]: I think you mentioned this isn't the exact format of the key and secret - do we want to highlight that somewhere?

File: hubspot-snowflake-aws-ml-insights/snowflake-queries/snowflake-to-s3-hs-contacts.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+2. Update the S3 Bucket URL based on the bucket created by the AWS Personalize ML OpsStack CloudFormation template
+3. Follow the documentation from Snowflake on unloading data from Snowflake to S3: https://docs.snowflake.com/en/user-guide/data-unload-s3
+Note: This query uses an AWS Secret Key tied to an AWS IAM User to authorize the unload into S3, not a Snowflake stage with a dedicated IAM user, which is Snowflake's recommended approach
+*/
+COPY INTO 's3://your-bucket-url/train/users.csv'
+FROM (
+    SELECT 
+        c.OBJECTID AS ID,
+        c.PROPERTY_EXTERNAL_ID AS USER_ID,
+        TO_VARCHAR(TO_NUMBER(c.PROPERTY_AGE)) AS AGE,
```
Comment (steven-miller) [inline hubspot-snowflake-aws-ml-insights/snowflake-queries/snowflake-to-s3-hs-contacts.sql:15]: I feel like I don't want to know 🥴

File: hubspot-snowflake-aws-ml-insights/snowflake-queries/snowflake-to-s3-hs-products.sql [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        interaction.PROPERTY_ITEM_ID as ITEM_ID,
+        interaction.PROPERTY_ITEM_CATEGORY_L1 as CATEGORY_L1,
+        interaction.PROPERTY_ITEM_CATEGORY_L2 as CATEGORY_L2,
+        REPLACE(interaction.PROPERTY_ITEM_DESCRIPTION, ',', ';') as PRODUCT_DESCRIPTION,
+        REPLACE(interaction.PROPERTY_ITEM_NAME, ',', ';') as PRODUCT_NAME,
+        TRY_CAST(interaction.PROPERTY_ITEM_PRICE AS STRING) as PRICE,
+        product.PROPERTY_GENDER as GENDER,
+        IFNULL(product.PROPERTY_PROMOTED, 'False') AS PROMOTED
+    FROM HUB_23826378_DB.V2_DAILY.EVENTS_AWS_E_COMMERCE_INTERACTIONS as interaction LEFT OUTER JOIN
+    HUB_23826378_DB.V2_DAILY.OBJECTS_LINE_ITEMS product ON product.PROPERTY_EXTERNAL_ID = interaction.PROPERTY_ITEM_ID
```
Comment (steven-miller) [inline hubspot-snowflake-aws-ml-insights/snowflake-queries/snowflake-to-s3-hs-products.sql:24]: You may want a brief explainer in the comments above on why we're using a `LEFT OUTER JOIN` - took me a minute to remember and figure out 😄 probably something like:
> We are retrieving all product user interactions and the relevant product information for those interactions.

**Review by steven-miller — APPROVED** [review]
👍 whoever wrote that last query needs a raise 😄

**Comment by robertainslie** [comment]
Added note, merging into the programmable-automation branch

**Pull request merged**